### PR TITLE
feat: Pass along env to proxy for substitution

### PR DESCRIPTION
**Pull request by johanandren**
Depends on protocol change in proxy https://github.com/lightbend/kalix-proxy/pull/1692

### CODE REVIEW

**Comment by johanandren** [comment]
Proxy now out in prod

File: sdk/java-sdk/src/main/resources/reference.conf [hunk 1]
```diff
@@ -16,13 +16,21 @@ kalix {
...
+    # The path to a file on the classpath that contains a compiled descriptor that contains source info.
+    # This descriptor should have been compiled by protobuf using the --descriptor_set_out and
+    # --include_source_info flags. If this file is present, the Java SDK will use this to fill in the source
+    # info for descriptors it sends to the proxy during discovery, since the descriptors compiled into the
+    # generated Java source do not include source info. To disable use of such a descriptor for locating
+    # source info, set this to "disabled".
     protobuf-descriptor-with-source-info-path = "/protobuf/descriptor-sets/user-function.desc"
+
+    # By default all environment variables of the process is passed along to the proxy, they are used only for
+    # substituion usage in the descriptor options such as topic names. By default all environment is passed along,
```
Comment (efgpinto) [inline sdk/java-sdk/src/main/resources/reference.conf:28]:
```suggestion
    # substitution usage in the descriptor options such as topic names.
```

Was "By default all environment is passed along," a leftover maybe?

File: sdk/java-sdk/src/main/scala/kalix/javasdk/impl/DiscoveryImpl.scala [hunk 2]
```diff
@@ -90,6 +82,27 @@ class DiscoveryImpl(
...
       in.protocolMajorVersion,
       in.protocolMinorVersion)
+
+    // possibly filtered or hidden env, passed along for substitution in descriptor options
+    val env: Map[String, String] = system.settings.config.getAnyRef("kalix.discovery.pass-along-env-allow") match {
+      case false => Map.empty
+      case true  => sys.env
+      case allowed: util.ArrayList[String @unchecked] =>
+        allowed.asScala.flatMap(name => sys.env.get(name).map(value => name -> value)).toMap
+    }
```
Comment (efgpinto) [inline sdk/java-sdk/src/main/scala/kalix/javasdk/impl/DiscoveryImpl.scala:92]: so, if the property gets configured wrongly (for instance, passing a string directly) this throws and discovery breaks I guess? User would know that way, right? or should we help them with a specific message maybe?

File: sdk/java-sdk/src/main/resources/reference.conf [hunk 3]
```diff
@@ -16,13 +16,21 @@ kalix {
...
-    // source info, set this to "disabled".
+    # The path to a file on the classpath that contains a compiled descriptor that contains source info.
+    # This descriptor should have been compiled by protobuf using the --descriptor_set_out and
+    # --include_source_info flags. If this file is present, the Java SDK will use this to fill in the source
+    # info for descriptors it sends to the proxy during discovery, since the descriptors compiled into the
+    # generated Java source do not include source info. To disable use of such a descriptor for locating
+    # source info, set this to "disabled".
     protobuf-descriptor-with-source-info-path = "/protobuf/descriptor-sets/user-function.desc"
+
+    # By default all environment variables of the process is passed along to the proxy, they are used only for
```
Comment (efgpinto) [inline sdk/java-sdk/src/main/resources/reference.conf:27]:
```suggestion
    # By default all environment variables of the process are passed along to the proxy, they are used only for
```

**Review by efgpinto — APPROVED** [review]
LGTM, some comments about tiny details.

**Pull request merged**